### ISSUE TITLE
Tx propagation sometimes failing after Dandelion++ timeout

### ISSUE THREAD

**Issue by selsta** [monero-project/monero#6929]
I’ve read multiple reports about this issue and also encountered this myself.

Sent a transaction, it got stuck in the mempool, even after waiting 10+ minutes. In the case of Dandelion++ timeout the transaction should get propagated to all nodes, this was not the case this time. I had 64 out peers so it is unlikely that all of them dropped the transaction.

`relay_tx` did also not help in this case. I tried multiple times and also waited. Only solution was to flush_txpool and resend the transaction.

These reports have been happening since v0.17

**Comment by selsta** [comment]
@vtnerd Any idea?

**Comment by LocalMonero** [comment]
> `relay_tx` did also not help in this case. I tried multiple times and also waited. Only solution was to flush_txpool and resend the transaction.
> 
> These reports have been happening since v0.17

We've had this happen since v0.16

**Comment by moneromooo-monero** [comment]
When running "print_pool_sh, in monerod in such a case, what are the values of "receive_time" and "relayed" ?

What is the value of "date +%s" on the machine running monerod ?

**Comment by vtnerd** [comment]
I'll have to look into some of the changes made since v0.16. Its likely either `tx_pool.cpp` or `levin_notify.cpp`. The latter was unit tested a bit more directly. If it happens again, try to get a printouts of the pool (with the hashes scrubbed).

**Comment by LocalMonero** [comment]
@moneromooo-monero @vtnerd 

`id: xxx
blob_size: 1682
weight: 2219
fee: 0.000099420000
fee/byte: 0.000000044803
receive_time: 1603953989 (10 hours ago)
relayed: 1603983131 (2 hours ago)
do_not_relay: F
kept_by_block: F
double_spend_seen: T
max_used_block_height: xxx
max_used_block_id: xxx
last_failed_height: 0
last_failed_id: 0000000000000000000000000000000000000000000000000000000000000000
`

**Comment by vtnerd** [comment]
@LocalMonero did you have i2p and/or tor enabled via `--tx-proxy` ?

**Comment by LocalMonero** [comment]
@vtnerd nope.

**Comment by canlin05** [comment]
Any update on this?  Unable to send anything.  Have tried connecting to remote node node.xmr.to on port 18081 but all transactions are failing.

**Comment by selsta** [comment]
@canlin05 is this with GUI? node xmr to should definitely work, how often did you try?

**Comment by canlin05** [comment]
Trying for 3rd time now ... still waiting on confirmation at 12min.  It has failed sooner in previous attempts, so hoping it goes through this time.  Though, 12min seems a long time without a confirmation (?)

**Comment by canlin05** [comment]
And yes, it is with GUI.

**Comment by selsta** [comment]
You can check if your transaction is in the mempool here: https://xmrchain.net/

12 minutes without a block is nothing unusual.

**Comment by canlin05** [comment]
Getting some confirmations now.  Thanks for the quick replies, btw.  Should I use this remote node all the time in the future?

**Comment by selsta** [comment]
We will release a new GUI version tomorrow that should fix simple mode again. So you can choose what you want to use.

**Comment by Gingeropolous** [comment]
@selsta , this issue is closed, right?

**Comment by selsta** [comment]
@Gingeropolous the bug was never found so I assume no

**Comment by xanoni** [comment]
Happens quite often to me that the TX doesn't confirm after an hour or more. I haven't identified the pattern, yet. I use 

Settings (see also https://github.com/monero-project/monero/issues/7863):

```
tx-proxy=i2p,127.0.0.1:4447,25
tx-proxy=tor,127.0.0.1:9050,25

anonymous-inbound=XYZ.b32.i2p:${I2P_EXT_PORT},127.0.0.1:18080,25 # `i2pd` tunnel was defined manually
anonymous-inbound=XYZ.onion:${TOR_EXT_PORT},127.0.0.1:18080,25

# settings equivalent to the below also used for RPC 
p2p-ignore-ipv4=1
p2p-use-ipv6=0
p2p-bind-ip=127.0.0.1
p2p-bind-port=18080

hide-my-port=1
public-node=0
no-igd=1
allow-local-ip=0
no-zmq=1
```

Not sure if `relay_tx` helps (last time it didn't). Seems flushing the TX and resending it has been my best bet to date.

**Comment by selsta** [comment]
Should be solved in v0.18.0.0
